src/sentry/buffer/redis.py

```diff
@@ -263,19 +265,35 @@ def get_redis_connection(self, key: str, transaction: bool = True) -> Pipeline:
         pipe = conn.pipeline(transaction=transaction)
         return pipe
 
-    def _execute_redis_operation(self, key: str, operation: RedisOperation, *args: Any) -> Any:
+    def _execute_redis_operation(
+        self, key: str, operation: RedisOperation, *args: Any, **kwargs: Any
+    ) -> Any:
         pending_key = self._make_pending_key_from_key(key)
         pipe = self.get_redis_connection(pending_key)
-        getattr(pipe, operation.value)(key, *args)
+        getattr(pipe, operation.value)(key, *args, **kwargs)
         if args:
             pipe.expire(key, self.key_expire)
         return pipe.execute()
 
-    def push_to_set(self, key: str, value: list[int] | int) -> None:
-        self._execute_redis_operation(key, RedisOperation.SET_ADD, value)
+    def push_to_sorted_set(self, key: str, value: list[int] | int) -> None:
+        self._execute_redis_operation(key, RedisOperation.SORTED_SET_ADD, {value: time()})
```

Comment:
I think we should also pass the `GT` option here so that we only ever increase the score https://redis.io/docs/latest/commands/zadd/

Reply:
Which Redis version are we using? I can't get this to work and that's my only guess why:
> Starting with Redis version 6.2.0: Added the GT and LT options.

I tried `gt=True` and `GT=True` and always get `TypeError: Redis.zadd() got an unexpected keyword argument 'gt'`

Reply:
Yeah I think you're right - at the very least the client we're using doesn't support it. Not a huge deal